pkg/action/dependency.go

```diff
@@ -43,25 +44,66 @@ func NewDependency() *Dependency {
 	return &Dependency{}
 }
 
-// List executes 'helm dependency list'.
-func (d *Dependency) List(chartpath string, out io.Writer) error {
-	c, err := loader.Load(chartpath)
+type dependencyListElement struct {
+	Name       string `json:"name"`
+	Version    string `json:"version"`
+	Repository string `json:"repository"`
+	Status     string `json:"status"`
+}
+
+type DependencyListWriter struct {
+	Chartpath string
+}
+
+func (w *DependencyListWriter) WriteTable(out io.Writer) error {
```

Comment:
If you look at the existing places that write to JSON (e.g., history, repo listing, etc) it is in the `cmd/helm` package. It's part of the output of the CLI. If there isn't a reason to break from that pattern, can this be refactored to use the existing pattern.

Reply:
As I wrote in the PR description, I tried to keep making as many minor changes as possible to ease the code review process. I saw the need for these changes myself, but I was afraid it would cause a lot of changes to it which would increase the size of the PR. 
I made the changes you requested, changed the tests to fit and added some informative comments where it was missing.